test/e2e/nodenetworkstate_test.go

```diff
@@ -0,0 +1,56 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1"
+)
+
+var _ = Describe("NodeNetworkState", func() {
+	Context("when desiredState is configured", func() {
+		var (
+			desiredState nmstatev1.State
+		)
+		JustBeforeEach(func() {
+			for _, node := range nodes {
+				updateDesiredState(namespace, node, desiredState)
```

Comment:
This is confusing. Combining operations performed in BeforeEach and JustBeforeEach gives me headache, jumping up and down. Cannot we just do all operations in BeforeEach directly? You can create a helper function to config desired state everywhere.

You could create onEachNode(func) helper that you could use in BeforeEach and It

Reply:
I was doing this ginkgo way, so we put the scenarios at BeforeEach, JustBeforeEach do just the stuff to updateDesiredState BeforeEach set the specific config, but we can replace it by helper.

Reply:
I don't like when I need to think too much while reading tests from top to bottom, but if it is **the** Ginkgo way, so be it.